src/intersection.js

```diff
@@ -23,5 +23,7 @@ var uniq = require('./uniq');
  *      R.intersection([1,2,3,4], [7,6,5,4,3]); //=> [4, 3]
  */
 module.exports = _curry2(function intersection(list1, list2) {
-  return uniq(_filter(flip(_contains)(list1), list2));
+  var lookupList = list1.length > list2.length ? list1 : list2;
+  var filteredList = lookupList == list1 ? list2 : list1;
```

Comment:
I would not want to ship with Babel as a dependency. We would have to build the output ourselves.

I very much like having the shipped source being as clean as possible, noting that `ramda.js` is our flagship project and should always be as readable and clean as possible.

I wasn't worried about performance.

@scott-christopher [recently demonstrated][sc] that at least with rollup, we can build very clean packages, one of the things that has bothered me about babel and webpack.


  [sc]: https://github.com/ramda/ramda/issues/1505#issuecomment-165610371